spring-cloud-app-broker-acceptance-tests/src/test/java/org/springframework/cloud/appbroker/acceptance/UpdateInstanceWithNewServiceAcceptanceTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.appbroker.acceptance;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.cloudfoundry.operations.applications.ApplicationSummary;
+import org.cloudfoundry.operations.services.ServiceInstance;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.cloud.appbroker.acceptance.services.ConditionalReplaceBackingService.REPLACEMENT_BACKING_SERVICE_PARAMETER_KEY;
+
+class UpdateInstanceWithNewServiceAcceptanceTest extends CloudFoundryAcceptanceTest {
+
+	private static final String APP_NAME = "app-update-with-new-services";
+
+	private static final String SI_NAME = "si-update-with-new-services";
+
+	private static final String OLD_BACKING_SI_NAME = "backing-service-instance-old";
+
+	private static final String NEW_BACKING_SI_NAME = "backing-service-instance-new";
+
+	private static final String SUFFIX = "update-with-new-services";
+
+	private static final String APP_SERVICE_NAME = "app-service-" + SUFFIX;
+
+	private static final String BACKING_SERVICE_NAME = "backing-service-" + SUFFIX;
+
+
+	@Autowired
+	private HealthListener healthListener;
+
+	@Override
+	protected String testSuffix() {
+		return SUFFIX;
+	}
+
+	@Override
+	protected String appServiceName() {
+		return APP_SERVICE_NAME;
+	}
+
+	@Override
+	protected String backingServiceName() {
+		return BACKING_SERVICE_NAME;
+	}
+
+	@Test
+	@AppBrokerTestProperties({
+		"spring.cloud.appbroker.services[0].service-name=" + APP_SERVICE_NAME,
+		"spring.cloud.appbroker.services[0].plan-name=" + PLAN_NAME,
+
+		"spring.cloud.appbroker.services[0].apps[0].name=" + APP_NAME,
+		"spring.cloud.appbroker.services[0].apps[0].path=" + BACKING_APP_PATH,
+		"spring.cloud.appbroker.services[0].apps[0].parameters-transformers[0].name=ConditionalBackingAppUpdate",
```

Comment:
Excellent test @LittleBaiBai! 
But wondering, as mentioned above, if instead of having a parameter-transformer to do the service part wouldn't it be better if we had a workflow as part of this class.